lib/mapper_class.py

```diff
@@ -1,51 +1,273 @@
+        if self.verbose>0:
             print("--->creating mapper graphs...")
+
         self._runMapper()
 
         # assign train points to graph node bins
-        if log.level == logging.DEBUG:
+        if self.verbose>0:
             print("--->assigning train points to graph node bins...")
         self._makeGraphBins()
 
         return self.total_graphbinm
 
     def _getLatentRep(self):
-        self.rep = getLatentRep(self.n_components, self.data, remake=self.remake)
+
+        if not self.remake and os.path.exists("%s_latent" % self.label):
+            frin = open(TEMP_DATA + "%s_latent" % self.label, "rb")
+            self.rep = pickle.load(frin)
+            return
+
+        if self.verbose>0:
+            print("------> fitting {}-component PCA to data of shape {}...".format(self.n_components, self.data.shape))
+        pca = MyPCA(n_components=self.n_components)
+        pca.fit(self.data)
+
+        self.rep = pca #LatentRep([pca], self.label)
+        fr = open(TEMP_DATA + "%s_latent" % self.label, "wb")
+        pickle.dump(self.rep, fr)
+        fr.close()
 
     def _runMapper(self):
-        self.graphs, self.mapper_pipes = runMapper(self.n_components, self.data, self.rep, remake=self.remake)
+        if not self.remake and os.path.exists("%s_firstsimplegap_graphs" % self.label):
+            fgin = open(TEMP_DATA + "%s_firstsimplegap_graphs" % self.label, "rb")
+            self.graphs = pickle.load(fgin)
+
+            fpin = open(TEMP_DATA + "%s_mapper_pipes" % self.label, "rb")
+            self.mapper_pipes = pickle.load(fpin)
+            return
+
+        clusterer = FirstSimpleGap()
+        self.mapper_pipes = []
+        pca = self.rep
+
+        if self.verbose>0:
+            print("------> creating projection components...")
+
+        for k in range(self.n_components):
+            if self.verbose>0:
+                print("---------> on component {}/{}...".format(k + 1, self.n_components))
+            proj = Projection(columns=k)
+            filter_func = Pipeline(steps=[('pca', pca), ('proj', proj)])
+            cover = OneDimensionalCover(n_intervals=self.n_intervals, overlap_frac=self.overlap_frac)
+            mapper_pipe = make_mapper_pipeline(scaler=None,
+                                               filter_func=filter_func,
+                                               cover=cover,
+                                               clusterer=clusterer,
+                                               verbose=(self.verbose > 0),
+                                               n_jobs=1)
+            mapper_pipe.set_params(filter_func__proj__columns=k)
+            self.mapper_pipes.append(("PCA%d" % (k + 1), mapper_pipe))
+
+        # try parallelization
+        if self.verbose>0:
+            print("------> entering parallelization...")
+        self.graphs = Parallel(n_jobs=5, prefer="threads")(
+            delayed(mapper_pipe[1].fit_transform)(self.data) for mapper_pipe in self.mapper_pipes
```

Comment:
the code to compute this delayed object was unchanged? That's why I'm not seeing it